### PR TITLE
Fix the sub-store validation to match snapd

### PR DESCRIPTION
**Pull request by slimjim777**

### CODE REVIEW

File: datastore/modeladapter_test.go [hunk 1]
```diff
@@ -43,17 +42,6 @@ func TestModelNameEmpty(t *testing.T) {
 	}
 }
 
-func TestModelNameInvalidChar(t *testing.T) {
```
Comment (rmescandon) [inline datastore/modeladapter_test.go]: This kind of changes deserve some tests

Comment (slimjim777) [reply datastore/modeladapter_test.go]: This test is no longer valid... that's why I removed it.

Comment (rmescandon) [reply datastore/modeladapter_test.go]: Sure, but I was pointing if another similar test would be needed like TestValidateCaseInsensitive()

Comment (slimjim777) [reply datastore/modeladapter_test.go]: As discussed, this is covered in TestModelNameUpperChar

**Pull request merged**